api/oc_network_events.c

```diff
@@ -179,6 +186,13 @@ oc_network_drop_receive_events(const oc_endpoint_t *endpoint)
     message = next;
   }
   oc_network_event_handler_mutex_unlock();
+
+ #ifdef OC_DYNAMIC_ALLOCATION 
+  if(oc_get_network_events_queue_length(endpoint->device) + dropped >= OC_MAX_NUM_CONCURRENT_REQUESTS) {
```

Comment:
you can reuse `oc_network_event_handler_mutex ` in previous lock by calling `oc_get_network_events_queue_length_locked(endpoint->device)`